### PR TITLE
React hooks update

### PR DESCRIPTION
**Pull request by ajbogh**
Updated all files except graph-view.js to use hooks. Code seems stable but will need testers to verify functionality.

This will be a breaking change requiring a major version bump.

Known issues:

- Layout engine doesn't re-render nodes and edges when some nodes are moved or interacted with, leading to an edge dangling or a node rendering offset from its true position.

### CODE REVIEW

File: src/components/node-text.js [hunk 1]
```diff
@@ -15,51 +15,83 @@
...
+    nodeTypes,
+  ]);
+
+  useEffect(() => {
+    d3.select(nodeTextRef.current).on('click', () => handleTextClick(d3.event));
+  }, [handleTextClick]);
+
+  return (
+    // Note: when "is" is used, we must use real HTML attributes, like text-anchor, not React-based attributes
+    <text ref={nodeTextRef} class={className} text-anchor="middle" is="text">
```
Comment (ksnyder9801) [inline src/components/node-text.js:79]: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/is
why are we using "is"? it seems to be meant for Web Component (which we're not using here) and breaks some stuff based on the comments

File: package.json [hunk 2]
```diff
@@ -113,8 +129,8 @@
     "lint": "eslint src",
     "lint-fix": "eslint --fix src",
     "precommit": "lint-staged && npm run test",
-    "prefast-test": "npm run prepublish",
-    "prepublish": "npm run package",
+    "prefast-test": "npm run prepublishOnly",
+    "prepublishOnly": "npm run package",
```
Comment (ksnyder9801) [inline package.json:133]: unrelated?

**Review by ksnyder9801 — COMMENTED** [review]
any idea on the dangling edge issue - is it actually a regression?

File: src/components/node-text.js (hunk 1 above)
Comment (ajbogh) [reply src/components/node-text.js:79]: Enzyme doesn't recognize SVG elements so you have to use "is" to trick it into rendering the element as a normal HTML element. You can write anything in the "is" value but the common use is to rewrite the element name itself.

```
Warning: The tag <use> is unrecognized in this browser. If you meant to render a React component, start its name with an uppercase letter.
```

https://stackoverflow.com/questions/48751919/how-can-i-suppress-the-tag-some-tag-is-unrecognized-in-this-browser-warning/55537927#55537927

File: package.json (hunk 2 above)
Comment (ajbogh) [reply package.json:133]: It suppresses a warning about feature deprecation in new versions of node. Not unrelated because the packages and the newer node version (12.4.0) was tested with this update.

**Comment by ajbogh** [comment]
> any idea on the dangling edge issue - is it actually a regression?

@ksnyder9801, I'm unsure if it's a regression, but I have noticed some glitches with the LayoutEngine features in the past. I will try to address this bug in this update.

One thing that I noticed is that the LayoutEngine feature creates a significant lag on initial render for large graphs, so I may need to look into that in the near future as well.

**Review by ksnyder9801 — APPROVED** [review]
